src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.ServiceClientAdapter/BMSAPIs/ItemAPIs.cs

```diff
@@ -136,9 +136,18 @@ public partial class ServiceClientAdapter
             string resourceGroupName = null)
         {
             BackupRequestResource triggerBackupRequest = new BackupRequestResource();
-            IaasVMBackupRequest iaasVmBackupRequest = new IaasVMBackupRequest();
-            iaasVmBackupRequest.RecoveryPointExpiryTimeInUTC = expiryDateTimeUtc;
-            triggerBackupRequest.Properties = iaasVmBackupRequest;
+            if (string.Compare(containerName.Split(';')[0], "IaasVMContainer") == 0)
```

Comment:
It is best to move this provider specific logic into the Provider layer itself, instead of comparing on the `containerName`. Can you please take `BackupRequest` as a parameter, create the `IaasVMBackupRequest` and `AzureFileShareBackupRequest` objects in their respective providers and pass them to this method?